packages/playwright-core/src/server/injected/injectedScript.ts

```diff
@@ -1611,7 +1611,9 @@ class ExpectedTextMatcher {
     }
   }
 
-  matches(text: string): boolean {
+  matches(text: string | string[]): boolean {
+    if (Array.isArray(text))
```

Comment:
I think it would be easier to follow if we introduce a separate `to.have.class.partial`, or branch on the `partial` option somewhere earlier.